src/Orchard/WebApi/Routes/StandardExtensionHttpRouteProvider.cs

```diff
@@ -22,9 +22,10 @@ public class StandardExtensionHttpRouteProvider : IHttpRouteProvider {
                 var displayPath = item.Distinct().Single();
 
                 yield return new HttpRouteDescriptor {
+                    Name = displayPath + ".Api",
                     Priority = -10,
                     RouteTemplate = "api/" + displayPath + "/{controller}/{id}",

```

Comment:
@sebastienros,

As I remember, only in the rare case where someone uses its own custom HttpRouteDescriptor, with a named route, with the same default name defined here `ModuleName.Api`, with a priority <= -10, and uses this route name in an urlhelper extension...

If so, you're right, he may have to change e.g the priority. But the program will not fail for a duplicate route name (fixed in this PR). And, because of the priority issue (fixed in this PR), most of the time when people define their own HttpRouteDescriptor, they use a priority > -10...

But, if you want, I can revert this change about default httproute names, and keep only the fixes about priority and duplicate route names... Let me know.

Best